yapapi/strategy.py

```diff
@@ -21,10 +23,130 @@
 SCORE_REJECTED: Final[float] = -1.0
 SCORE_TRUSTED: Final[float] = 100.0
 
+PROP_DEBIT_NOTE_INTERVAL_SEC: Final[str] = "golem.com.scheme.payu.debit-note.interval-sec?"
+PROP_PAYMENT_TIMEOUT_SEC: Final[str] = "golem.com.scheme.payu.payment-timeout-sec?"
+PROP_DEBIT_NOTE_ACCEPTANCE_TIMEOUT: Final[str] = "golem.com.payment.debit-notes.accept-timeout?"
+
+MID_AGREEMENT_PAYMENTS_PROPS = [PROP_DEBIT_NOTE_INTERVAL_SEC, PROP_PAYMENT_TIMEOUT_SEC]
+
+
+@dataclass
+class PropValueRange:
+    min: Optional[float] = None
+    max: Optional[float] = None
+
+    def __contains__(self, item: float) -> bool:
+        return (self.min is None or item >= self.min) and (self.max is None or item <= self.max)
+
+    def closest_acceptable(self, item: float):
+        if self.min is not None and item < self.min:
+            return self.min
+        if self.max is not None and item > self.max:
+            return self.max
+        return item
+
+    def __str__(self):
+        return f"({self.min}, {self.max})"
+
+
+DEFAULT_DEBIT_NOTE_ACCEPTANCE_TIMEOUT_SEC: Final[int] = 30
+DEFAULT_DEBIT_NOTE_INTERVAL_SEC: Final[int] = 60
+DEFAULT_PAYMENT_TIMEOUT_SEC: Final[int] = 18000
+MIN_EXPIRATION_FOR_MID_AGREEMENT_PAYMENTS: Final[int] = DEFAULT_PAYMENT_TIMEOUT_SEC
+
+DEBIT_NOTE_INTERVAL_GRACE_PERIOD: Final[int] = 30
+
+
+DEFAULT_PROPERTY_VALUE_RANGES: Dict[str, PropValueRange] = {
+    PROP_DEBIT_NOTE_INTERVAL_SEC: PropValueRange(DEFAULT_DEBIT_NOTE_INTERVAL_SEC, None),
+    PROP_PAYMENT_TIMEOUT_SEC: PropValueRange(DEFAULT_PAYMENT_TIMEOUT_SEC, None),
+    PROP_DEBIT_NOTE_ACCEPTANCE_TIMEOUT: PropValueRange(
+        DEFAULT_DEBIT_NOTE_ACCEPTANCE_TIMEOUT_SEC, None
+    ),
+}
+
+logger = logging.getLogger(__name__)
+
 
 class MarketStrategy(DemandDecorator, abc.ABC):
     """Abstract market strategy."""
 
+    acceptable_prop_value_ranges: Dict[str, PropValueRange]
+
+    def set_prop_value_ranges_defaults(
+        self, acceptable_prop_value_ranges: Dict[str, PropValueRange]
+    ) -> None:
+        try:
+            value_ranges = self.acceptable_prop_value_ranges
+            for key, value in acceptable_prop_value_ranges.items():
+                if key not in value_ranges:
+                    value_ranges[key] = value
+        except AttributeError:
+            self.acceptable_prop_value_ranges = acceptable_prop_value_ranges.copy()
+
+    async def respond_to_provider_offer(
```

Comment:
This function doesn't respond to anything, it just prepares a response.
So maybe `prepare_response_to_offer` or something like this? 

I do like the `CQRS` pattern - here we have a `Q` that is named as `C`, this is misleading.

Reply:
well, I'm going to argue that - within the context of the strategy - that _is_ a response to the provider's offer - so the strategy _does_ respond to the offer ... it's not relevant that the response is not _sent_ to the provider... it still produces a response...